### ISSUE TITLE
BeginMainMenu over other contents such as DockSpaceOverViewport

### ISSUE THREAD

**Issue by DarkLight-Ent** [ocornut/imgui#3035]
**Version/Branch of Dear ImGui:**

Version: v1.75
Branch:Docking

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_dx11.cpp 
Compiler: MS Visual Studio 2019
Operating System: Windows 10

**My Question:**
It seems BeginMainMenuBar() overlays rest of the ImGui Content

Visual feedback
![image](https://user-images.githubusercontent.com/60792979/74945611-1b703a00-5409-11ea-958b-2e3e4b8434bb.png)

**Here's my main menu code snippet**
This is my code for testing basic main menu
```
ImGui::BeginMainMenuBar();
		if (ImGui::BeginMenu("File"))
		{
			ImGui::MenuItem("New");
			ImGui::MenuItem("Create");
			ImGui::EndMenu();
		}

ImGui::EndMainMenuBar();
```
I couldnt tell if this is a bug,or im not passing a required flag to ImGui framework.

**Comment by ocornut** [comment]
Hello,

Your code snippet is missing the fundamental fact that you are creating a window with a dock-space that fills the screen. This is where something should be adjusted, ideally you'd have said window be considerate of the fact that there is a menu bar here.

There's currently no standardized way for them to communicate that to each others, but you can create your window right under. A menu bar is always `ImGui::GetFrameHeight()` high, so your fullscreen window could start at `viewport->Pos + ImVec2(0, ImGui::GetFrameHeight())` instead of `viewport->Pos`.

I would however like to get this problem solved in a more standardized and automatic manner, by providing a per-viewport mechanism of maintaining a "work area" similarly to what OS does.

**Comment by DarkLight-Ent** [comment]
Hello ocornut,

Thank you for your detailed reply,before i post this issue i searched for alternative ways to solve this behaviour before i present to you.I searched for a method which sets the window depth,however i never imagined the thing i looked indexed as FrameHeight in your framework how silly of me.

I suppose i have to adjust my frames according to mainmenu begin position and i wonder what approach you will pursue on this matter in near or far future.It's my personal observation that it can be done by layering depth per-viewport or per-window.I wonder if you could let me know this approach will be a roadblock in your current framework/progress or not

**Comment by ocornut** [comment]
I don't understand why you suggest introducing the concept of depth, when by definition we are trying to get those things to not overlay each others. Anyway I have a plan ahead to make this more automatic in function like `DockspaceOverViewport()`.

**Comment by ocornut** [comment]
This is now fixed by 75de34e281141f851f7e5f9f6a923573167bc0f0 !
